labs/a1.md

```diff
@@ -0,0 +1,139 @@
+---
+title: Advanced Lab 1
+layout: lab
+---
+
+Lecture: Thurs 2/1
+Released: Mon 2/5
+Due: End of Feb
+
+# Advanced section, Lab 1
+
+## Grading note
+Labs are graded on completion. Treat this lab as seeds of exploration instead of just a grade. If you don't pass on the first submission, you can have it checked off in-person by a decal facilitator.
+
+Since you know how to use unix tools (though you may be more or less familiar with certain tools), the goal of this lab is to drop you in the wilderness. You can find your way out! :D
+
+
+## Composability & workflows
+This lab can be done on your own UNIX-like machine, or you can ssh into `tsunami.ocf.berkeley.edu` using your OCF account to finish the lab there. As always, `man` and Google will be your friends.
+
+
+### Shell. web fetching, parsing, and frequency analysis.
+- [ ] Download Alice in Wonderland by parsing out a download link from Project Gutenberg.
+
+    https://www.gutenberg.org/ebooks/11
+
+As it turns out, Project Gutenberg doesn't like users of curl, and will demand you enable javascript.  It's probably a resonable defense against naive abuse of curl by people scraping the site, but it's annoying to deal with.
+
+
+You can just tell the server that you're Firefox instead:
+
+    curl -s -A "Mozilla/5.0 (X11; Linux i686) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/51.0.2704.79 Chrome/51.0.2704.79 Safari/537.36"  https://www.gutenberg.org/
+    
+Credit to: [gurditsbedi on medium](https://medium.com/@gurditsbedi/automating-download-from-project-gutenberg-using-the-linux-terminal-e9f35bbbd06c)
+
+
+
+- [ ] Isolate the body content from the headers and footers, perhaps saving to an intermediate file. 
+
+Hint: 
+`less` may be useful for inspection, and`sed` with no-echo option. 
+
+`sed`'s "#,#p" line-number/print argument (play with ranges, offset, and start/end of file chars...twist on regex line terminators).
+
+- [ ] Process the document to get word counts. You may need to clean up the data, especially punctuation. As with all text processing in the shell, Regex will be your friend (and worst enemy). Check out [regexr](https://regexr.com/) to help you construct proper regular expressions.
+
+(regex)
+
+`grep` has options to match whole words, and to print only match-contents (instead of a whole line containing a match). You need not use "regex-grouping", just a pattern. 
+
+`sed` may also be useful for filtering by whitelisting and/or blacklisting characters. In either case, character classes "[_]" and (defining them negatively) "[^_]" can help. In both cases, use, but beware the `-` dash character: it gets interpreted as a range and triggers cryptic errors when that range is invalid. `tr`, `cut`, `sort`, and `uniq` also may be
+useful.
```

Comment:
you may want to specify structurally that these two sentences are logically extensions of the previous "Process the document" point. Otherwise they seem kind of disconnected and out of place.